sdks/python/apache_beam/pipeline.py

```diff
@@ -299,6 +299,9 @@ def _replace_if_needed(self, original_transform_node):
               original_transform_node.full_label,
               original_transform_node.inputs)
 
+          replacement_transform_node.resource_hints = (
+              original_transform_node.resource_hints)
```

Comment:
Thanks for catching this. Would a factory method in AppliedPTransform be a better way to generate replacement in a way that replacement doesn't lose some important fields that will be defined later?

Also, should this preserve `envirnonment` and `annotations` as well?